src/TLuaInterpreter.cpp

```diff
@@ -5778,10 +5780,114 @@ int TLuaInterpreter::sendTelnetChannel102(lua_State* L)
+        }
+        if (!pItem->isPushDownButton()) {
+            return warnArgumentValue(L, __func__, QStringLiteral("item Id with %1 is not a push-down button").arg(id).toUtf8().constData());
+        }
+    }
+
+    if (argType == LUA_TSTRING) {
+        QString name = lua_tostring(L, 1);
+        if (name.isEmpty()) {
+            return warnArgumentValue(L, __func__, "item name must not be an empty string");
+        }
+        pItem = host.getActionUnit()->findAction(name);
+        if (!pItem) {
+            return warnArgumentValue(L, __func__, QStringLiteral("no button item with name '%1' found").arg(name).toUtf8().constData());
+        }
+        if (!pItem->isPushDownButton()) {
+            return warnArgumentValue(L, __func__, QStringLiteral("item with name '%1' is not a push-down button").arg(name).toUtf8().constData());
+        }
+    }
+
+    if (!pItem) {
+        // we'll get here if the (first) argument is NOT usable:
+        lua_pushfstring(L, "%s: bad argument #1 type (ID as number or name as string expected, got %s!)",
+            __func__, luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    lua_pushboolean(L, pItem->mButtonState);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setButtonState
+int TLuaInterpreter::setButtonState(lua_State* L)
+{
+    auto& host = getHostFromLua(L);
+    auto argType = lua_type(L, 1);
+    TAction* pItem = nullptr;
+    if (argType == LUA_TNUMBER) {
+        int id = lua_tonumber(L, 1);
+        if (id < 0) {
+            return warnArgumentValue(L, __func__, QStringLiteral("item Id (%1) invalid, it must be equal or greater than zero").arg(id).toUtf8().constData());
+        }
+        pItem = host.getActionUnit()->getAction(id);
+        if (!pItem) {
+            return warnArgumentValue(L, __func__, QStringLiteral("no button item with Id %1 found").arg(id).toUtf8().constData());
+        }
+        if (!pItem->isPushDownButton()) {
+            return warnArgumentValue(L, __func__, QStringLiteral("item Id with %1 is not a push-down button").arg(id).toUtf8().constData());
+        }
+    }
+
+    if (argType == LUA_TSTRING) {
+        QString name = lua_tostring(L, 1);
+        if (name.isEmpty()) {
+            return warnArgumentValue(L, __func__, "item name must not be an empty string");
+        }
+        pItem = host.getActionUnit()->findAction(name);
+        if (!pItem) {
+            return warnArgumentValue(L, __func__, QStringLiteral("no button item with name '%1' found").arg(name).toUtf8().constData());
+        }
+        if (!pItem->isPushDownButton()) {
+            return warnArgumentValue(L, __func__, QStringLiteral("item with name '%1' is not a push-down button").arg(name).toUtf8().constData());
+        }
+    }
+
+    if (!pItem) {
+        // we'll get here if the (first) argument is NOT usable:
+        lua_pushfstring(L, "%s: bad argument #1 type (ID as number or name as string expected, got %s!)",
```

Comment:
See there is an "ID", let's use that consistently, shall we?

Reply:
Okay, in the scheme of things I can't be ~~arsed~~*bothered* to fight this...